pkg/asset/manifests/azure/cluster.go

```diff
@@ -267,6 +267,9 @@ func getNextAvailableIP(ctx context.Context, installConfig *installconfig.Instal
 	if err != nil {
 		return "", fmt.Errorf("failed to get azure ip availability: %w", err)
 	}
+	if availableIP == nil {
+		return "", fmt.Errorf("failed to get available IP in given machine network: %w", err)
+	}
 	if *availableIP.Available {
```

Comment:
I think `availableIP.Available` is also a pointer. We could also check that too, just to be safe?